pkg/jobrunaggregator/releasebigqueryloader/cmd.go

```diff
@@ -0,0 +1,104 @@
+package releasebigqueryloader
+
+import (
+	"context"
+	"net/http"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+
+	"github.com/openshift/ci-tools/pkg/jobrunaggregator/jobrunaggregatorlib"
+)
+
+type BigQueryReleaseUploadFlags struct {
+	DataCoordinates *jobrunaggregatorlib.BigQueryDataCoordinates
+	Authentication  *jobrunaggregatorlib.GoogleAuthenticationFlags
+	Releases        []string
+}
+
+func NewBigQueryReleaseUploadFlags() *BigQueryReleaseUploadFlags {
+	return &BigQueryReleaseUploadFlags{
+		DataCoordinates: jobrunaggregatorlib.NewBigQueryDataCoordinates(),
+		Authentication:  jobrunaggregatorlib.NewGoogleAuthenticationFlags(),
+	}
+}
+
+func (f *BigQueryReleaseUploadFlags) BindFlags(fs *pflag.FlagSet) {
+	f.DataCoordinates.BindFlags(fs)
+	f.Authentication.BindFlags(fs)
+	fs.StringArrayVar(&f.Releases, "releases", f.Releases, "openshift releases to collect data from")
+}
+
+func NewBigQueryReleaseUploadFlagsCommand() *cobra.Command {
+	f := NewBigQueryReleaseUploadFlags()
+
+	cmd := &cobra.Command{
+		Use:          "upload-releases",
+		Long:         `Upload release data to bigquery`,
+		SilenceUsage: true,
+
+		RunE: func(cmd *cobra.Command, args []string) error {
+			ctx := context.Background()
+
+			if err := f.Validate(); err != nil {
+				logrus.WithError(err).Fatal("Flags are invalid")
+			}
+			o, err := f.ToOptions(ctx)
+			if err != nil {
+				logrus.WithError(err).Fatal("Failed to build runtime options")
+			}
+
+			if err := o.Run(ctx); err != nil {
+				logrus.WithError(err).Fatal("Command failed")
+			}
+
+			return nil
+		},
+
+		Args: jobrunaggregatorlib.NoArgs,
+	}
+
+	f.BindFlags(cmd.Flags())
+
+	return cmd
+}
+
+// Validate checks to see if the user-input is likely to produce functional runtime options
+func (f *BigQueryReleaseUploadFlags) Validate() error {
+	if err := f.DataCoordinates.Validate(); err != nil {
+		return err
+	}
+	if err := f.Authentication.Validate(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// ToOptions goes from the user input to the runtime values need to run the command.
+// Expect to see unit tests on the options, but not on the flags which are simply value mappings.
+func (f *BigQueryReleaseUploadFlags) ToOptions(ctx context.Context) (*allReleaseUploaderOptions, error) {
+	// Create a new GCS Client
+	gcsClient, err := f.Authentication.NewCIGCSClient(ctx, "origin-ci-test")
```

Comment:
reading top to bottom, I'm surprised you need a GCS client

Reply:
I'm surprised too... I'll remove it